src/Text.Analyzers/UnitTests/IdentifiersShouldBeSpelledCorrectlyTests.cs

```diff
@@ -1,8 +1,531 @@
+        [InlineData("class Program { void Member(string name) { } }")]
+        [InlineData("class Program { delegate int GetNumber(string name); }")]
+        [InlineData("class Program<TResource> { }")]
+        public async Task NoMisspellings_Verify_NoDiagnostics(string source)
+        {
+            await VerifyCSharpAsync(source);
+        }
+
+        [Fact]
+        public async Task MisspellingAllowedByGlobalXmlDictionary_Verify_NoDiagnostics()
+        {
+            var source = "class Clazz { }";
+            var dictionary = CreateXmlDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(source, dictionary);
+        }
+
+        [Fact]
+        public async Task MisspellingAllowedByGlobalDicDictionary_Verify_NoDiagnostics()
+        {
+            var source = "class Clazz { }";
+            var dictionary = CreateDicDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(source, dictionary);
+        }
+
+        [Fact]
+        public async Task MisspellingsAllowedByMultipleGlobalDictionaries_Verify_NoDiagnostics()
+        {
+            var source = @"class Clazz { const string Naem = ""foo""; }";
+            var xmlDictionary = CreateXmlDictionary(new[] { "clazz" });
+            var dicDictionary = CreateDicDictionary(new[] { "naem" });
+
+            await VerifyCSharpAsync(source, new[] { xmlDictionary, dicDictionary });
+        }
+
+        [Fact]
+        public async Task CorrectWordDisallowedByGlobalXmlDictionary_Verify_EmitsDiagnostic()
+        {
+            var source = "class Program { }";
+            var dictionary = CreateXmlDictionary(null, new[] { "program" });
+
+            await VerifyCSharpAsync(
+                source,
+                dictionary,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.TypeRule)
+                    .WithLocation(1, 7)
+                    .WithArguments("Program", "Program"));
+        }
+
+        [Fact]
+        public async Task MisspellingAllowedByProjectDictionary_Verify_NoDiagnostics()
+        {
+            var source = "class Clazz {}";
+            var dictionary = CreateDicDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(source, dictionary);
+        }
+
+        [Fact(Skip = "Adding additional files to specific projects is not yet supported")]
+        public async Task MisspellingAllowedByDifferentProjectDictionary_Verify_EmitsDiagnostic()
+        {
+            var source = "class Clazz {}";
+            var dictionary = CreateDicDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(
+                source,
+                dictionary,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.TypeRule)
+                    .WithLocation(1, 7)
+                    .WithArguments("Clazz", "Clazz"));
+        }
+
+        [Fact(Skip = "Specifying assembly names is not yet supported")]
```

Comment:
Hum I think we can do it, pretty sure I did it for some test. Let me try to find back the test.

Reply:
Did you find the test?

Reply:
Nevermind; found an example